### PR TITLE
Android bootstrap and log callback registration

### LINKED ISSUE #68: Adapt logs to work with Android

### PR DESCRIPTION
**Pull request by ivnsch**

### CODE REVIEW

File: src/simple_logger.rs [hunk 1]
```diff
@@ -9,18 +9,18 @@ use std::sync::{mpsc::Sender, Once};
...
-    Trace,
-    Debug,
-    Info,
-    Warn,
-    Error,
+    Trace = 0,
+    Debug = 1,
+    Info = 2,
+    Warn = 3,
+    Error = 4,
```
Comment (ivnsch) [inline src/simple_logger.rs:23]: Explicit is better when logic depends on the values.

File: src/android/android_interface.rs [hunk 2]
```diff
@@ -1 +1,155 @@
...
+    // We need JNIEnv to call the callback.
+    // JNIEnv is valid only in the same thread, so we have to store the vm instead, and use it to get
+    // a JNIEnv for the current thread.
+    // See https://developer.android.com/training/articles/perf-jni#javavm-and-jnienvb
+    java_vm: JavaVM,
+}
+
+impl LogCallbackWrapper for LogCallbackWrapperImpl {
+    fn call(&self, level: CoreLogLevel, text: String) {
+        let env = self.java_vm.attach_current_thread().unwrap();
```
Comment (duskoo) [inline src/android/android_interface.rs:113]: Multiple threads can be attached to VM?

File: src/android/android_interface.rs [hunk 3]
```diff
@@ -1 +1,155 @@
-// TODO
+use crate::{errors::ServicesError, init_db, simple_logger};
```
Comment (duskoo) [inline src/android/android_interface.rs:1]: This file is Android specific, so it should be compiled conditionally: only when target_os is android?

**Review by duskoo — APPROVED** [review]

File: src/android/android_interface.rs (hunk 2 above)
Comment (ivnsch) [reply src/android/android_interface.rs:113]: This returns a `JNIEnv` for the current thread. `JNIEnv` is valid only in the thread where it's created. Not sure about the internals.

File: src/android/android_interface.rs (hunk 3 above)
Comment (ivnsch) [reply src/android/android_interface.rs:1]: It's declared in android/mod.rs, which is imported conditionally in lib.rs

**Pull request merged**